src/NuGet.Clients/NuGet.SolutionRestoreManager/ISolutionRestoreChecker.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using NuGet.Commands;
+using NuGet.ProjectModel;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// An up to date checker for a solution.
+    /// </summary>
+    public interface ISolutionRestoreChecker
```

Comment:
Do you see any benefits in making the interface methods async ?

Reply:
There are no heavy/async invocations happening right now. 
I had them async at first, but I was calling Task.FromResult pretty much everywhere. 

Given that this is a type inside NuGet.Clients we are free to change as we see fit, so for now, I think they should remain sync. 

If we add some extra parallelism to the check we can try to make them async.